### PR TITLE
Add install test for cloud targets

### PR DESCRIPTION
**Pull request by hcho3**
Test whether the x86 DLR wheel works on C4, C5, M4, M5 cloud targets.

### CODE REVIEW

File: src/dlr.cc [hunk 1]
```diff
@@ -184,14 +184,12 @@ void DLRModel::SetupTVMModule(const std::string& model_path) {
 void DLRModel::SetupTreeliteModule(const std::string& model_path) {
   ModelPath paths = get_treelite_paths(model_path);
   int num_worker_threads = -1; // use the maximum amount of threads
-  int include_master_thread = 1;
```
Comment (ziyu-guo) [inline src/dlr.cc]: What is it for?

Comment (hcho3) [reply src/dlr.cc]: This corresponds to the API change in the upstream. The `include_master_thread` parameter was causing confusion and thus removed in dmlc/treelite#79

File: Jenkinsfile [hunk 2]
```diff
@@ -88,3 +102,21 @@ def AMD64BuildGPU() {
     }
   }
 }
+
+def CloudInstallAndTest(cloudTarget) {
+  def nodeReq = "ubuntu && amd64 && ${cloudTarget}"
+  node(nodeReq) {
+    echo "Installing DLR package for ${cloudTarget} target"
+    withAWS(credentials:'Neo-AI-CI-Fleet') {
+      files = s3FindFiles(bucket: 'neo-ai-dlr-jenkins-artifacts', glob: "${env.JOB_NAME}/${env.BUILD_ID}/artifacts/dlr-*-manylinux1_x86_64.whl")
```
Comment (wweic) [inline Jenkinsfile]: How are these files in `neo-ai-dlr-jenkins-artifacts` generated?

Comment (hcho3) [reply Jenkinsfile]: In the Build stage

File: Jenkinsfile [hunk 3]
```diff
@@ -3,6 +3,11 @@
 // Jenkins pipeline
 // See documents at https://jenkins.io/doc/book/pipeline/jenkinsfile/
 
+// List of cloud targets
+def cloudTargetMatrix = [
+  "c4", "c5", "m4", "m5"
```
Comment (wweic) [inline Jenkinsfile]: why not also add `p2` `p3`?

File: Jenkinsfile (hunk 2 above)
Comment (wweic) [reply Jenkinsfile]: Got it. Thank you!

File: Jenkinsfile (hunk 3 above)
Comment (hcho3) [reply Jenkinsfile]: Let’s wait until the EC2 instance limit is increased?

**Comment by hcho3** [comment]
I now realize that history re-writing invalidates submodule links in previous history. From now on, I will do periodic merge with upstream, not rebase

File: Jenkinsfile (hunk 3 above)
Comment (wweic) [reply Jenkinsfile]: Make sense.

**Pull request merged**